app/controllers/items_controller.rb

```diff
@@ -18,10 +20,34 @@ def create
     end
   end
 
+  #def show
+    #@item = Item.find(params[:id])
+  #end
+
+  def edit
+  end
+
+  def update
+    if @item.update(item_params)
+      redirect_to action: :show
+    else
+      render :edit
+    end
+  end
+
   private
 
   def item_params
     params.require(:item).permit(:item_name, :description, :category_id, :status_id, :charge_id, :prefecture_id,
                                  :shipping_date_id, :price, :image).merge(user_id: current_user.id)
   end
+
+  def move_to_index
+    redirect_to root_path unless user_signed_in? && current_user.id == @item.user_id
+  end
```

Comment:
```ruby
redirect_to root_path unless user_signed_in? && current_user.id == @item.user_id
```
`unless user_signed_in?`ですが、
deviseのヘルパーメソッドであるauthenticate_user!を使用することで同じ処理が可能となります。
authenticate_user!内にupdateアクションを追加し、こちらのメソッドは削除しておきましょう。

ご不明点がございましたら、コネクトでご質問ください。

Reply:
以前の方と言っていることが違っていて、困ります。

Reply:
次に確認してくださる方、もうこちらに関してツッコミを入れないでください。

Reply:
こちらの言葉不足で、大変申し訳ございません。

こちらの意図とは別の形で伝わっている可能性がございますので、
細かく説明させていただきます。

```suggestion
  def move_to_index
    redirect_to root_path unless current_user.id == @item.user_id
  end
```
move_to_indexメソッドを削除するのではなく、user_signed_in?を削除していただければ幸いです。
ご不明点がございましたら、再度コメントいただきますようよろしくお願いいたします。